src/com/chrome/apis/polymer.js

```diff
@@ -0,0 +1,207 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+foam.CLASS({
+  package: 'foam.u2.md',
+  name: 'StackView',
+  extends: 'foam.u2.BasicStackView',
+  methods: [
+    function initE() {
+      this.cssClass('flex').cssClass('layout').cssClass('vertical');
+      this.SUPER();
+    }
+  ]
+});
+
+foam.CLASS({
+  package: 'foam.u2.md',
+  name: 'Toolbar',
+  extends: 'foam.u2.Element',
+  properties: [
+    {
+      name: 'title'
+    },
+    {
+      name: 'leftActions_',
+      factory: function() {
+        return [];
+      }
+    },
+    {
+      name: 'rightActions_',
+      factory: function() {
+        return [];
+      }
+    },
+    [ 'nodeName', 'paper-toolbar' ]
+  ],
+
+  methods: [
+    function initE() {
+      this.add(this.slot('leftActions_'));
+      this.start('span').cssClass('title').add(this.title$).end();
+      this.add(this.slot('rightActions_'));
+    },
+
+    function addLeftAction(button) {
+      this.leftActions_.push(button);
+      this.propertyChange.pub('leftActions_', this.leftActions_);
+    },
+    function addRightAction(button) {
+      this.rightActions_.push(button);
+      this.propertyChange.pub('rightActions_', this.rightActions_);
+    }
+  ]
+});
+
+foam.CLASS({
+  package: 'foam.u2.md',
+  name: 'ToolbarContainer',
+  extends: 'foam.u2.Element',
+
+  properties: [
+    {
+      name: 'toolbar',
+      required: true
+    },
+    {
+      name: 'body',
+      factory: function() {
+        return this.E('div');
+      }
+    },
+    [ 'nodeName', 'paper-header-panel' ]
+  ],
+
+  methods: [
+    function initE() {
+      this.attrs({ fullbleed: true });
+      this.add(this.toolbar.cssClass('paper-header'));
+      this.add(this.body.cssClass('content'));
+    }
+  ]
+});
+
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'PolymerWrapper',
+  extends: 'foam.u2.View'
+});
+
+foam.CLASS({
+  package: 'foam.u2.md',
+  name: 'StringRefinement',
+  refines: 'foam.core.String',
+  imports: [
+    'setTimeout'
+  ],
+  properties: [
+    {
+      class: 'Boolean',
+      name: 'onKey',
+      value: 'false'
+    },
+    {
+      name: 'toPropertyE',
+      value: function(X) {
+        var e = X.lookup('foam.u2.PolymerWrapper').create({
+          nodeName: 'paper-input'
+        }, X);
+
+        e.attrs({ label: this.label$ });
+
+        var event = this.onKey ? 'keypress' : 'change';
+        var self = this;
+
+        e.onload.sub(function() {
```

Comment:
How is that different than regular U2 Elements?  They also aren't ready to add listeners to until they've been loaded, so U2 remembers the listeners and then adds them once it is loaded.  I still don't see why this is needed.  Can't the PolymerWrapper handle this?